src/core/StringReader.hpp

```diff
@@ -27,6 +27,7 @@ interface IStringReader
     virtual bool TryRead(codepoint_t * outCodepoint)       abstract;
     virtual void Skip()                                    abstract;
     virtual bool CanRead()                           const abstract;
+    virtual ~IStringReader() = default;
```

Comment:
Move to top and put a line break after it like I did for other interfaces.
What does default mean again? Is it same as `{ }`?